app/views/account_requests/new.html.erb

```diff
@@ -6,14 +6,30 @@
       </div>
 
       <div class="form-check">
-        <%= radio_button_tag(:account, :bank, false, class: 'form-check-input account_type') %>
+        <%= radio_button_tag(:account, :bank, flash.present?, class: 'form-check-input account_type') %>
```

Comment:
Wouldn't this always select this value if _any_ previous value was selected?

In Rails, usually you would use a model to store the information and then re-render the page with the model rather than using the flash storage to try and figure out if the form was already filled out or not. Here's a simple example: https://github.com/rubyforgood/human-essentials/blob/6a9b9d518c755a88217a32f32d6aaed935d294f3/app/controllers/partners/families_controller.rb#L57-L57

Reply:
👋 Hi @dorner , 

Thanks for your patience - I've been quite busy the past few weeks.

My understanding is that these radio buttons were built outside of the form as a way to only display the account_request model's form if a user selected that they are indeed an Essentials bank. I believe its purpose is to help redirect partner agencies to the proper login page and prevent erroneous diaper bank accounts. As a result,` render :new ` does not automatically re-select the previous radio button 😢 

In this case, errors should only occur from the Essentials Bank form. So in its current state, it is necessary to re-select the Essentials Bank radio button once those errors do occur in order to show the form with its errors (resolving the original issue).

I have pushed up changes for the script to automatically select the Essentials Bank radio button when there are errors present. This will display the form with its errors and preserve that radio button's checked field as false.

Please let me know if you think there is a better way to accomplish this or if you would like me to refactor the flow of selecting Partner vs. Essentials Bank. I appreciate your time!

Reply:
I'd probably add another parameter to the request indicating that it's coming from that workflow rather than relying on the presence or absence of errors.

Reply:
Sounds good - thank you! I will look into implementing a new parameter to try to take care of that instead. 

For my knowledge (if you have time to answer), is there a specific reason we would want to avoid relying on the presence of errors in this case?

Reply:
Mainly because errors could happen for different reasons. If we ever change the flow, it's very unintuitive that we treat the form differently if there is or isn't an error. Basically it violates the Principle of Least Surprise :)